.github/workflows/push_pr.yml

```diff
@@ -83,6 +83,34 @@ jobs:
         run: |
           .\build\windows\unit_tests.ps1
 
+  # can't run this step inside of container because of tests specific
+  test-integration-nix:
```

Comment:
I'm wondering wether it could be possible to place this yaml block within its own workflow file and share `on`  triggers with the prerelease one, so we avoid duplicating the same block...

Reply:
Sure that could be an option, but will we do the same for the snyk, test and all the other jobs?

Reply:
For other pipelines we don't run tests on prerelease (other than the ones made by make), since we assume a responsible developer (heh) that won't release from an untested branch. We could discuss whether is a good approach or not, but I think it's nice to be consistent.

Reply:
I think it's a good way to avoid duplicating workflow code.

Tradeoff could be: workflow gets more scattered, as for instance a PR workflow lays within several files, but each file has a single responsibility.

Reply:
I'm not strong on this opinion, but if asked I'd rather running tests on pre-release as sometimes stuff is just pushed without flowing throw PR.

Reply:
I'm leaving this conversation here, as solution is fine enough and conversation goal goes beyond the PR scope.